### PR TITLE
Implemented Assignment questions Day 1 on iterators

### PR DESCRIPTION
**Pull request by shravan-cc**
1. This PR introduces an iterator object named alphabetIterator that iterates over the English alphabet from 'a' to 'z'. It includes methods to make the object iterable and supports iteration through while loop, for...of loop, and conversion to an array.
2. The PR also includes test cases to cover different aspects of the alphabetIterator functionality, including iteration through while loop, for...of loop, and conversion to an array.

### CODE REVIEW

File: iterator.test.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+/**
+ * Test suite for the iterator functionality.
+ */
+describe("Assignment questions test on iterators", () => {
+  /**
+   * Test case for iterating through alphabets using a while loop.
+   */
+  test("Test for iterating through alphabets using while loop", () => {
+    const iterator = alphabetIterator[Symbol.iterator]();
+    let expectedCharCode = 97;
```
Comment (shripada) [inline iterator.test.js]: We can avoid hardcoding by syaing `'a'.codePointAt(0)`

File: iterator.js [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  [Symbol.iterator]() {
+    let index = 0;
+    return {
+      /**
+       * Method of the iterator object. It returns the next value in the iteration sequence.
+       * @returns {Object} An object containing the value of the current iteration and a boolean indicating whether the iteration is complete.
+       */
+      next() {
+        if (index < 26) {
+          const value = String.fromCharCode(97 + index);
```
Comment (shripada) [inline iterator.js]: We can avoid hardcoding by syaing 'a'.codePointAt(0)

File: iterator.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+   */
+  [Symbol.iterator]() {
+    let index = 0;
+    return {
+      /**
+       * Method of the iterator object. It returns the next value in the iteration sequence.
+       * @returns {Object} An object containing the value of the current iteration and a boolean indicating whether the iteration is complete.
+       */
+      next() {
+        if (index < 26) {
```
Comment (shripada) [inline iterator.js]: can simply check if we generated 'z'.  and avoid index.

**Review by shripada — CHANGES_REQUESTED** [review]

File: iterator.test.js (hunk 1 above)
Comment (shravan-cc) [reply iterator.test.js]: Addressed the changes.

File: iterator.js (hunk 2 above)
Comment (shravan-cc) [reply iterator.js]: Addressed the changes.

File: iterator.js (hunk 3 above)
Comment (shravan-cc) [reply iterator.js]: Addressed the changes.

**Review by shripada — APPROVED** [review]

**Pull request merged**